FBShimmering/FBShimmeringLayer.m

```diff
@@ -244,6 +244,22 @@ - (void)setShimmeringOpacity:(CGFloat)shimmeringOpacity
   }
 }
 
+- (CFTimeInterval)shimmeringBeginTime
+{
+  return [_maskLayer animationForKey:kFBShimmerSlideAnimationKey].beginTime;
+}
+
+- (void)setShimmeringBeginTime:(CFTimeInterval)beginTime
+{
+  CAAnimation *slideAnimation = [[_maskLayer animationForKey:kFBShimmerSlideAnimationKey] mutableCopy];
+  if (nil == slideAnimation) {
+    return;
+  }
+
+  slideAnimation.beginTime = beginTime;
+  [_maskLayer addAnimation:slideAnimation forKey:kFBShimmerSlideAnimationKey];
```

Comment:
Rather than adding this here, set an ivar and read that out inside the shimmer update function?